packages/core/src/anchor/ethereum/ethereum-anchor-validator.ts

```diff
@@ -137,32 +173,70 @@ export class EthereumAnchorValidator implements AnchorValidator {
   }
 
   /**
-   * Validate anchor proof on the chain
+   * Validate version 0 anchor proof on the chain by the reading tx data directly
    * @param anchorProof - Anchor proof instance
    */
-  async validateChainInclusion(anchorProof: AnchorProof): Promise<void> {
+  async parseAnchorProofV0(anchorProof: AnchorProof): Promise<ValidationResult> {
     const decoded = decode(anchorProof.txHash.multihash.bytes)
     const txHash = '0x' + uint8arrays.toString(decoded.digest, 'base16')
-
     const [transaction, block] = await this._getTransactionAndBlockInfo(anchorProof.chainId, txHash)
     const txValueHexNumber = parseInt(transaction.data, 16)
     const rootValueHexNumber = parseInt('0x' + anchorProof.root.toString(base16), 16)
+    return { txResponse: transaction, block, txValueHexNumber, rootValueHexNumber }
+  }
+
+  /**
+   * Validate version 1 anchor proof on the chain by parsing first encoded parameter
+   * @param anchorProof - Anchor proof instance
+   */
+  async parseAnchorProofV1(anchorProof: AnchorProof): Promise<ValidationResult> {
+    const decoded = decode(anchorProof.txHash.multihash.bytes)
+    const txHash = '0x' + uint8arrays.toString(decoded.digest, 'base16')
+    const [transaction, block] = await this._getTransactionAndBlockInfo(anchorProof.chainId, txHash)
+    const decodedArgs = iface.decodeFunctionData('anchor', transaction.data)
+    const rootCID = decodedArgs[0]
+    const txValueHexNumber = parseInt(rootCID, 16)
+    const rootValueHexNumber = parseInt('0x' + anchorProof.root.toString(base16), 16)
+    return { txResponse: transaction, block, txValueHexNumber, rootValueHexNumber }
+  }
 
-    if (txValueHexNumber !== rootValueHexNumber) {
+  async validate(anchorProof: AnchorProof): Promise<ValidationResult> {
```

Comment:
this isn't really validating anything, it's just parsing the information from the transaction.  Can we call it `parseTransactionData` or similar?

Reply:
or I guess `parseAnchorProof` for symmetry with its internal helpers